task-forage.lic

```diff
@@ -371,29 +368,25 @@ class TaskForage
 
   def use_boosts
     echo 'use_boosts' if @debug
-    fput('info') # Check to see if a boost is already active
-    pause 0.25
 
-    lines = reget(50).reverse
-    lines.each do |line|
-      case line
-      when / +Task Master, expiring /
-        echo 'Already boosted' if @debug
+    # Check to see if a boost is already active
+    case DRC.bput('info', ' +Task Master, expiring ', ' +Encumbrance :', { 'timeout' => 0.5, 'suppress_no_match' => true })
```

Comment:
might bump that timer to 2 or 3 to account for lag. I usually don't go lower than 5 for that reason.

Reply:
Yeah, I don't love forcing a lengthy(ish) pause for the good folks with reliable internet connections.  But in hindsight it's probably the thing to do given that boosts are extremely finite, and double-redeeming one because of a missed match is wasteful. I can live with 2 sec.

Reply:
Oh, while considering this I realized my switch from reget to bput was flawed (bput options need to come before matching patterns, which was obscured by the fact that the Encumbrance always match first, and return a "no boost" condition).  I'll do some better testing and submit.

Reply:
I think most people who decide to use boosts will just set the option to true, burn through whatever backlogged boosts they have, and leave the option enabled to consume additional boosts whenever they get earned again. Assuming that 99% of the time there will be be no boosts active, that's a 1-5 second waste each time the script is run (depending on timeout). That might end up being less efficient than just never using boosts at all.

So, a lot of rambling for a pretty minor decision, but I prefer the super-short timeout (0.5 sec) to minimize that, recognizing that lag could cause a false negative on the check. If that does happen, no boost will be activated that particular run, which is not a big deal IMO.